rom/laghos_rom.cpp

```diff
@@ -4647,35 +4660,45 @@ void ROM_Operator::ForceIntegratorEQP(Vector & res,
 		trial_fe->CalcShape(ip, eshape);
 		unitE = 1.0;
 
+		const int eos = elemIndex * nvdof;
+
 		if (energy_conserve)
 		{
 			// energy-conserving EQP
 			for (int j = 0; j < rdim; ++j)
 			{
-				// TODO: v_e should be the jth velocity basis vector DOFs that
-				// correspond to the current element.
-				// I think the below is not right.
-				basis->GetBasisVectorV(false, j, v_e);	
+				// v_e: jth V basis vector's DOFs on this element
+				for (int k = 0; k < nvdof; ++k) v_e[k] = W_elems(eos + k, j);
```

Comment:
Here I access matrix `W_elems`, which has the data of `Wmat` organized by element, to get the V basis vector DOFs corresponding to the current element. 

This is similar to the way it's done in the basic EQP case (where, however, the basis vector is multiplied by Mv^{-1}) so that there's minimal changes to the code.

I do the calculation the same way for the energy equation.